openquake/commonlib/valid.py

```diff
@@ -159,6 +159,21 @@ def __call__(self, value):
 
 name_with_dashes = Regex(r'^[a-zA-Z_][\w\-]*$')
 
+MAX_ID_LENGTH = 100
+
+
+def simple_id(value):
+    """
+    Check the source id; the only accepted chars are a-zA-Z0-9_-
+    """
+    if len(value) > MAX_ID_LENGTH:
+        raise ValueError('The ID %r is longer than %d character' %
+                         (value, MAX_ID_LENGTH))
+    if re.match(r'^[\w_\-]+$', value):
+        return value
+    raise ValueError(
+        'The ID %r contains invalid characters' % value)
```

Comment:
It would be better for the error message to indicate explicitly which characters are valid (i.e. a-z A-Z 0-9 _ and -)